cumulusci/robotframework/pageobjects/BasePageObjects.py

```diff
@@ -0,0 +1,270 @@
+# Note: Unlike in traditional robot libraries, this docstring will end
+# up in output from the libdoc task. For that to happen, it must be
+# before the imports.
+
+"""
+The following page objects are automatically included when
+importing the PageObject library. You should not directly import this
+file.
+
+"""
+
+import re
+from cumulusci.robotframework.pageobjects import pageobject
+from cumulusci.robotframework.pageobjects import BasePage
+from cumulusci.robotframework.utils import capture_screenshot_on_error
+
+# This will appear in the generated documentation in place of
+# the filename.
+TITLE = "Base Page Objects"
+
+
+@pageobject(page_type="Listing")
+class ListingPage(BasePage):
+    """Page object representing a Listing page
+
+    When going to the Listing page, you need to specify the object name. You
+    may also specify the name of a filter.
+
+    Example
+
+    | Go to page  Listing  Contact  filterName=Recent
+
+    """
+
+    def _go_to_page(self, filter_name=None):
+        url_template = "{root}/lightning/o/{object_name}/list"
+        url = url_template.format(
+            root=self.cumulusci.org.lightning_base_url, object_name=self.object_name
+        )
+        if filter_name:
+            url += "?filterName={}".format(filter_name)
+        self.selenium.go_to(url)
+        self.salesforce.wait_until_loading_is_complete()
+
+    def _is_current_page(self):
+        self.selenium.location_should_contain(
+            "/lightning/o/{}/list".format(self.object_name)
+        )
+
+
+@pageobject("New")
+class NewDialog(BasePage):
+    """A page object representing the New Object dialog
```

Comment:
These are called Modals in the Lightning Design System (https://www.lightningdesignsystem.com/components/modals/) so I would use that term throughout rather than dialog.